### DISCUSSION TITLE
Update/Upsert sandbox records map on spefic field other than Id

### DISCUSSION THREAD

**Discussion by FabLeroux** [tprouvot/Salesforce-Inspector-reloaded#376] [Ideas]
Hi,

Our use case : 
Now and then we need to update some records on our sandbox
The records don't have always the same id than the source org. 
It'd be great if'd able to map the record within the two org on an other field than the id

Would you mind considering this for future feature ?

**Comment by tprouvot** [comment]
Hi @FabLeroux,
This is the perfect use case for external Id, do you have any on this object ?

**Comment by FabLeroux** [comment]
Hi @tprouvot 

in fact the specific case on which we face an issue is  that we create new users on our main org and on our training org in parallel
with time and some new features we need to update attributes in user records which obviously  don't have the same ids and no external id
we would like to use email as pivot field

**Comment by tprouvot** [comment]
Humm I can see the challenge you are facing with this particular use case, I'll think about it and try to find a way to address it.

I know that [sfdmu](https://help.sfdmu.com/get-started) can use normal fields as external id (such as email) but it requires to do some queries to 'premap' the id field. I'll take a look and check if we could use the same logical

**Comment by tprouvot** [comment]
@FabLeroux,
Just tried something with the extension without any modification and it seems to be working 🤯 

<img width="1008" alt="image" src="https://github.com/tprouvot/Salesforce-Inspector-reloaded/assets/35368290/778aba23-1dee-4635-8b4a-2ed33b21bb8d">

Let me know if you also have a success

**Comment by FabLeroux** [comment]
My bad, you're right!
So not using Update but prefer Upsert function  and especially check the value of 'External Id' which is by default at 'Id' and can be change with specific text and not only with one of the picklist value displayed when click on arrow.
Merci

**Comment by tprouvot** [comment]
I think this is only specific to Email which can be considered as externalId (don't know why at this point 😄 ) 
But you can use Email as externalId and it should be ok for your use case 

<img width="491" alt="image" src="https://github.com/tprouvot/Salesforce-Inspector-reloaded/assets/35368290/0c9b68c8-bd19-466f-9778-95e4f0abe914">

**Comment by tprouvot** [comment]
I could figured why !
https://ideas.salesforce.com/s/idea/a0B8W00000Gdq7JUAR/enable-user-email-to-be-used-as-an-external-id?sfdcIFrameOrigin=null
